docs/java-rest/high-level/java-builders.asciidoc

```diff
@@ -0,0 +1,26 @@
+[[java-rest-high-java-builders]]
+== Using Java Builders
+
+The Java High Level REST Client depends on the Elasticsearch core project which provides
+two types of Java `Builders` objects:
```

Comment:
I don't know if we should say there are only two types of builders, after all search also has Suggestion builder, Sort, Highlightingbuilders etc... We don't need to mention them all here but maybe we can say these two are the most important ones?